vault/datadog_checks/vault/vault.py

```diff
@@ -98,7 +99,7 @@ def check(self, _):
             for submit_function in submission_queue:
                 submit_function(tags=tags)
 
-        if self._client_token:
+        if self._client_token or self._no_token:
```

Comment:
In the exception, should we check if `self._no_token is True` and log a message when it fails so we know a token may be required?

Reply:
I think the exception re-raise + service check is sufficient, wdyt?